### PR TITLE
vmu: Add function vmu_draw_lcd_rotated()

### PR DESCRIPTION
**Pull request by pcercuei**
This function is equivalent to vmu_draw_lcd, but the image is rotated
180° so that the first byte of the bitmap corresponds to the top-left
corner, instead of the bottom-right one.

### CODE REVIEW

**Comment by gyrovorbis** [comment]
Assembly optimized VMU rendering? Hell yeah!

File: kernel/arch/dreamcast/include/dc/math.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+*/
+
+/**
+    \brief  Returns the bit-reverse value of the argument (where MSB
+            becomes LSB and vice-versa).
+    \return the bit-reverse value of the argument.
+*/
+unsigned int bit_reverse(unsigned int value);
+
+#endif /* __DC_RBIT_H */
```
Comment (QuzarDC) [inline kernel/arch/dreamcast/include/dc/math.h:27]: __DC_MATH_H

Comment (pcercuei) [reply kernel/arch/dreamcast/include/dc/math.h:27]: Good catch - the file was called rbit.h before I decided to just move it to a generic "math.h" header where other functions can be added later.

**Comment by pcercuei** [comment]
Note that I notice now that the `vmu_draw_lcd_rotated` function requires the bitmap array to be aligned to 4 bytes, because it's rotating the image 32 bits at a time. I don't know if that's a problem or not.

File: kernel/arch/dreamcast/include/dc/maple/vmu.h [hunk 2]
```diff
@@ -99,6 +99,21 @@ int vmu_beep_raw(maple_device_t * dev, uint32 beep);
 */
 int vmu_draw_lcd(maple_device_t * dev, void *bitmap);
 
+/** \brief  Display a 1bpp bitmap on a VMU screen.
+
+    This function sends a raw bitmap to a VMU to display on its screen. This
+    bitmap is 1bpp, and is 48x32 in size. This function is equivalent to
+    vmu_draw_lcd, but the image is rotated 180° so that the first byte of the
+    bitmap corresponds to the top-left corner, instead of the bottom-right one.
+
```
Comment (gyrovorbis) [inline kernel/arch/dreamcast/include/dc/maple/vmu.h:108]:
```suggestion

\warning
This function is optimized by an assembly routine which operates on 32 bits at a time. As such, the given bitmap must be 4-byte aligned. 

```

**Comment by gyrovorbis** [comment]
> Note that I notice now that the `vmu_draw_lcd_rotated` function requires the bitmap array to be aligned to 4 bytes, because it's rotating the image 32 bits at a time. I don't know if that's a problem or not.

I don't mind, but I've been going through the codebase documenting any side-effects, limitations, or requirements such as this using doxygen "warnings" to make sure the user knows what the deal is.

File: kernel/arch/dreamcast/hardware/maple/vmu.c [hunk 3]
```diff
@@ -206,6 +207,17 @@ int vmu_draw_lcd(maple_device_t * dev, void *bitmap) {
...
+int vmu_draw_lcd_rotated(maple_device_t *dev, const void *bitmap)
+{
+	uint32 bitmap_inverted[48];
+	unsigned int i;
+
+	for (i = 0; i < 48; i++)
+		bitmap_inverted[i] = bit_reverse(((uint32 *)bitmap)[47 - i]);
+
+	return vmu_draw_lcd(dev, bitmap_inverted);
+}
```
Comment (ljsebald) [inline kernel/arch/dreamcast/hardware/maple/vmu.c:219]:
```suggestion
int vmu_draw_lcd_rotated(maple_device_t *dev, const void *bitmap) {
    uint32 bitmap_inverted[48];
    unsigned int i;

    for (i = 0; i < 48; i++) {
        bitmap_inverted[i] = bit_reverse(((uint32 *)bitmap)[47 - i]);
    }

    return vmu_draw_lcd(dev, bitmap_inverted);
}
```

**Review by ljsebald — CHANGES_REQUESTED** [review]
See comments above.

**Review by QuzarDC — APPROVED** [review]

**Review by ljsebald — APPROVED** [review]

**Pull request merged**